src/ocean/ocean_idealised.f90

```diff
@@ -0,0 +1,180 @@
+MODULE ocean_idealised
+
+  ! Idealised ocean models
+
+! ===== Preamble =====
+! ====================
+
+  USE precisions                                             , ONLY: dp
+  USE mpi_basic                                              , ONLY: par, sync
+  USE control_resources_and_error_messaging                  , ONLY: crash, init_routine, finalise_routine, colour_string
+  USE model_configuration                                    , ONLY: C
+  USE parameters
+  USE mesh_types                                             , ONLY: type_mesh
+  USE ice_model_types                                        , ONLY: type_ice_model
+  USE ocean_model_types                                      , ONLY: type_ocean_model
+
+  IMPLICIT NONE
+
+CONTAINS
+
+! ===== Main routines =====
+! =========================
+
+  SUBROUTINE run_ocean_model_idealised( mesh, ice, ocean)
+    ! Calculate the ocean
+    !
+    ! Use an idealised ocean scheme
+
+    IMPLICIT NONE
+
+    ! In/output variables:
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_ice_model),                   INTENT(IN)    :: ice
+    TYPE(type_ocean_model),                 INTENT(INOUT) :: ocean
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'run_ocean_model_idealised'
+
+    ! Add routine to path
+    CALL init_routine( routine_name)
+
+    ! Run the chosen idealised ocean model
+    IF (C%choice_ocean_model_idealised == 'ISOMIP') THEN
+      ! No need to do anything 
+    ELSEIF (C%choice_ocean_model_idealised == 'TANH') THEN
+      ! No need to do anything 
+    ELSE
+      CALL crash('unknown choice_ocean_model_idealised "' // TRIM( C%choice_ocean_model_idealised) // '"')
+    END IF
```

Comment:
I want to eventually replace these IF, ELSEIF lists with SELECT CASE constructs (as they explicitly restrict you to have a single variable that determines the behaviour), with the crash as the first case under CASE DEFAULT. I still forget it myself too sometimes so a bit hypocritical, but still.

Reply:
Right, I'll change it here and keep it in mind for future 'cases'. Perhaps good to do a clean sweep through the code at some point for consistency then.